internal/dag/serviceapis_processor.go

```diff
@@ -26,5 +35,126 @@ func (p *ServiceAPIsProcessor) Run(dag *DAG, source *KubernetesCache) {
 	p.dag = dag
 	p.source = source
 
-	// NOT IMPLEMENTED
+	// reset the processor when we're done
+	defer func() {
+		p.dag = nil
+		p.source = nil
+	}()
+
+	for _, route := range p.validHTTPRoutes() {
+		p.computeHTTPRoute(route)
+	}
+}
+
+func (p *ServiceAPIsProcessor) computeHTTPRoute(route *serviceapis.HTTPRoute) {
+
+	// Determine the hosts on the route, if no hosts
+	// are defined, then set to "*".
+	var hosts []string
+	if len(route.Spec.Hostnames) == 0 {
+		hosts = append(hosts, "*")
+	} else {
+		for _, host := range route.Spec.Hostnames {
+			hosts = append(hosts, string(host))
+		}
+	}
+
+	for _, rule := range route.Spec.Rules {
+
+		// Paths are a set of paths that are
+		var pathPrefixes []string
+		var services []*Service
+
+		for _, match := range rule.Matches {
+			switch match.Path.Type {
+			case serviceapis.PathMatchPrefix:
+				pathPrefixes = append(pathPrefixes, stringOrDefault(match.Path.Value, "/"))
+			default:
+				p.Error("NOT IMPLEMENTED: Only PathMatchPrefix is currently implemented.")
+			}
+		}
+
+		for _, forward := range rule.ForwardTo {
+			// Verify the service is valid
+			if forward.ServiceName == nil {
+				p.Error("ServiceName must not specified and is currently only implemented!")
+				break
+			}
+			meta := types.NamespacedName{Name: *forward.ServiceName, Namespace: route.Namespace}
+
+			// TODO: Refactor EnsureService to take an int32 so conversion to intstr is not needed.
+			service, err := p.dag.EnsureService(meta, intstr.FromInt(int(forward.Port)), p.source)
```

Comment:
Note that if https://github.com/kubernetes-sigs/service-apis/pull/538 merges, port will be optional. With this PR, when port is unspecified, the implementation should forward the request with the port unmodified. Take a look and comment if this is an issue.

Reply:
#3352